src/utils/x-user-token.ts

```diff
@@ -0,0 +1,60 @@
+import { errorsToReadableMessages } from "@pagopa/ts-commons/lib/reporters";
+import * as E from "fp-ts/lib/Either";
+import { Either } from "fp-ts/lib/Either";
+import * as O from "fp-ts/lib/Option";
+import { Option } from "fp-ts/lib/Option";
+import { flow, pipe } from "fp-ts/lib/function";
+
+import { UserIdentity } from "../../generated/io-auth/UserIdentity";
+import { FF_IO_X_USER_TOKEN_ENABLED } from "../config";
+
+const parseUser = (value: string): Either<Error, UserIdentity> =>
+  pipe(
+    E.parseJSON(value, E.toError),
+    E.chain(
+      flow(
+        UserIdentity.decode,
+        E.mapLeft((err) => new Error(errorsToReadableMessages(err).join("/")))
+      )
+    )
+  );
+
+const decodeApiKey = (apiKey: string): Either<Error, UserIdentity> =>
+  pipe(
+    E.tryCatch(
+      () => Buffer.from(apiKey, "base64").toString("utf-8"),
+      (err) => E.toError(err)
+    ),
+    E.chain(
+      flow(
+        O.fromNullable,
+        E.fromOption(() => new Error("User not found")),
+        E.chain(parseUser)
+      )
```

Comment:
`toString` should never return a nullable value, at least an empty string.
```suggestion
      flow(
        E.fromPredicate(
          (decodedToken) => decodedToken !== "",
          () => new Error("Invalid token")
        ),
        E.chain(parseUser)
      )
```